projects/Mallard/src/download-edition/clear-issues-and-editions.ts

```diff
@@ -0,0 +1,148 @@
+        )
+        console.log(`Error cleaning up download issues folder `, error)
+        errorService.captureException(error)
+    }
+}
+
+export const deleteIssue = async (localId: string): Promise<boolean> => {
+    const issuePath = FSPaths.issueRoot(localId)
+    const doesItExist = await RNFS.exists(issuePath)
+    if (doesItExist) {
+        await RNFS.unlink(issuePath).catch(e => {
+            errorService.captureException(e)
+        })
+        localIssueListStore.remove(localId)
+        return true
+    }
+    return false
+}
+
+const deleteIssueFiles = async (): Promise<void> => {
+    await RNFS.unlink(FSPaths.issuesDir)
+    localIssueListStore.reset()
+    await prepFileSystem()
+    await clearDownloadsDirectory()
+}
+
+const deleteIssues = (issuesToDelete: string[], trackingId: PushTrackingId) => {
+    return Promise.all(
+        issuesToDelete.map((issue: string) => deleteIssue(issue)),
+    )
+        .then(() => pushTracking(trackingId, 'completed', Feature.CLEAR_ISSUES))
+        .catch(e => {
+            errorService.captureException(e)
+        })
+}
+
+const clearOldIssues = async (): Promise<void> => {
+    const edition = await getSelectedEditionSlug()
+    const files = await getLocalIssues(edition)
+
+    const iTD: string[] = await issuesToDelete(files)
+    return deleteIssues(iTD, 'clearOldIssues')
+}
+
+const editionDirsToClean = (
+    directoryList: { name: string; path: string }[],
+    editionList: EditionId[],
+): { name: string; path: string }[] => {
+    // we never want to delete default regional editions
+    const editionsToKeep = editionList.concat(
+        defaultRegionalEditions.map(e => e.edition),
+    )
+    // don't clean hidden folders, the download folder or editions we want to keep
+    return directoryList.filter(
+        d =>
+            !d.name.startsWith('.') &&
+            d.name !== 'download' &&
+            !editionsToKeep.includes(d.name),
+    )
+}
+
+const deleteOldEditionIssues = async (editionIds: EditionId[]) => {
+    const rootFolders = await RNFS.readDir(FSPaths.issuesDir)
+
+    const rootEditionFoldersToClean = editionDirsToClean(
+        rootFolders,
+        editionIds,
+    )
+
+    // we've had issues in the past with deleting issue folders, so just delete the issues for the edition
+    const issuesToClear = await Promise.all(
+        rootEditionFoldersToClean.map(e => getLocalIssues(e.name)),
+    )
+    issuesToClear.forEach(issues => {
+        try {
+            deleteIssues(issues, 'clearOldEditions')
+        } catch (error) {
+            errorService.captureException(error)
+        }
+    })
+}
+
+const cleanEditionsDownloadFolder = async (editionIds: EditionId[]) => {
+    const downloadFolders = await RNFS.readDir(FSPaths.downloadRoot)
+    const downloadEditionFoldersToDelete = editionDirsToClean(
+        downloadFolders,
+        editionIds,
+    )
+    // we don't care about download folders - delete them all!
```

Comment:
This comment confuses me a little since `editionDirsToClean` should never return the download folder. 
Would you mind adding a test to explain this?

Reply:
ah yeah this is confusing, I'll tidy it up and see if I can add some tests

Reply:
I had a look at this and have restructured the code a bit so there's less duplication and better comments.

Unfortunately editionDirsToClean is the only function that's easy to test without mocking out RNFS. I could do this, but maybe without the confusing comments this is now a bit more readable and we can get away without it?!